src/plugins/splitLargeMessages.ts

```diff
@@ -0,0 +1,167 @@
+                } else {
+                    chunks.push(line);
+                }
+            }
+            break;
+        case SplitPlace.LastCharacter:
+            chunks = [];
+            for (let idx = 0; idx < messageContent.length; idx += maxMessageSize) {
+                console.log(idx);
+                chunks.push(messageContent.slice(idx, idx + maxMessageSize));
+            }
+            break;
+    }
+    chunks = chunks.map(e => e.trimStart().trimEnd());
+    return chunks;
+}
+
+function sendMessage(channelId: BigInt | string, messageContent: string) {
+    const message = {
+        invalidEmojis: [],
+        tts: false,
+        validNonShortcutEmojis: [],
+        content: messageContent
+    };
+    MessageCreator.sendMessage(channelId, message, void 0);
+}
+
+async function sendMessages(channelId: BigInt | string, messageContents: string[], delay: number) {
+    for (const messageContent of messageContents) {
+        sendMessage(channelId, messageContent);
+        await wait(delay);
+    }
+}
+
+const settings = definePluginSettings({
+    splitPlace: {
+        description: "Where the message is split.",
+        type: OptionType.SELECT,
+        options: [
+            {
+                label: "On newline, falls back onto the other method if a line is too long. (default)",
+                value: SplitPlace.Newline,
+                default: true,
+            },
+            {
+                label: "On last possible character, crams everything into smallest amount of possible messages.",
+                value: SplitPlace.LastCharacter,
+            }
+        ],
+    },
+    delay: {
+        description: "Delay between messages in seconds, lower on your own risk.",
+        type: OptionType.SLIDER,
+        markers: makeRange(0, 1, 0.1),
+        default: 0.2,
+        stickToMarkers: false
+    }
+});
+
+export default definePlugin({
+    name: "Split Large Messages",
+    description: "Splits larger than allowed messages into chunks and sends them in seperate messages.",
+    authors: [
+        {
+            id: 174134334628823041n,
+            name: "Pièrre",
+        }
+    ],
+    settings,
+    patches: [
+        {
+            find: ".MESSAGE_TOO_LONG",
+            replacement: {
+                match: /;if\(\i\.length>\i\){\i&&null(?<=function\s\i\((\i)\).+?;if\((\i\.length>\i)\)({\i&&null))/,
+                replace: ";if(($2)&&($1.type.analyticsName!=\"normal\"))$3"
+            }
+        },
+        {
+            find: ".PREMIUM_MESSAGE_LENGTH_CHATBOX_FLAIR.format",
+            replacement: {
+                match: /,{children:\w+}\)\)}}\)]}\)(?<=function\s\w+\((\w+)\){var \w+,(\w+)=\w+\.type,(\w+)=\w+.textValue[.\s\S]+?,{children:(\w+)}\)\)}}\)]}\))/,
+                replace: ",{children:$2.analyticsName===\"normal\"?$3.length:$4}))}})]})"
+            }
+        }
+    ],
+    start() {
+        Vencord.Api.MessageEvents.addPreSendListener((channelId, message, _extra) => {
```

Comment:
avoid using the Vencord global, import the functions you need
```suggestion
        addPreSendListener((channelId, message, _extra) => {
```